src/main/java/com/github/platymemo/alaskanativecraft/mixin/client/InGameHudMixin.java

```diff
@@ -23,40 +18,19 @@
 
 @Environment(EnvType.CLIENT)
 @Mixin(InGameHud.class)
-public class InGameHudMixin {
-
+public abstract class InGameHudMixin {
+    private static final Identifier SNOWGOOGLES_BLUR = new Identifier(AlaskaNativeCraft.MOD_ID, "textures/misc/gogglesblur.png");
+    
     @Shadow
     @Final
     private MinecraftClient client;
-
-    @Shadow
-    private int scaledHeight;
-
-    @Shadow
-    private int scaledWidth;
-
+    
+    @Shadow protected abstract void renderOverlay(Identifier texture, float opacity);
+    
     @Inject(at = @At(value = "JUMP", opcode = Opcodes.IFEQ, ordinal = 1), method = "render")
     private void renderSnowGogglesOverlay(MatrixStack matrices, float tickDelta, CallbackInfo ci) {
         if (this.client.options.getPerspective().isFirstPerson() && this.client.player.getInventory().getArmorStack(3).getItem() == AlaskaItems.SNOW_GOGGLES) {
-            //TODO: test
-            RenderSystem.disableDepthTest();
-            RenderSystem.depthMask(false);
-            RenderSystem.defaultBlendFunc();
-            RenderSystem.setShaderColor(1.0F, 1.0F, 1.0F, 1.0F);
-            //RenderSystem.disableAlphaTest();
-            this.client.getTextureManager().bindTexture(new Identifier(AlaskaNativeCraft.MOD_ID, "textures/misc/gogglesblur.png"));
-            Tessellator tessellator = Tessellator.getInstance();
-            BufferBuilder bufferBuilder = tessellator.getBuffer();
-            bufferBuilder.begin(VertexFormat.DrawMode.QUADS, VertexFormats.POSITION_TEXTURE);
-            bufferBuilder.vertex(0.0D, this.scaledHeight, -90.0D).texture(0.0F, 1.0F).next();
-            bufferBuilder.vertex(this.scaledWidth, this.scaledHeight, -90.0D).texture(1.0F, 1.0F).next();
-            bufferBuilder.vertex(this.scaledWidth, 0.0D, -90.0D).texture(1.0F, 0.0F).next();
-            bufferBuilder.vertex(0.0D, 0.0D, -90.0D).texture(0.0F, 0.0F).next();
-            tessellator.draw();
-            RenderSystem.depthMask(true);
-            RenderSystem.enableDepthTest();
-            //RenderSystem.enableAlphaTest();
-            RenderSystem.setShaderColor(1.0F, 1.0F, 1.0F, 1.0F);
+            this.renderOverlay(SNOWGOOGLES_BLUR, 1.0F);
```

Comment:
Oh that is beautiful, I'm so glad 1.17 has this now

Reply:
Yeah, that is a great simplification, and makes total sense for the update as they've added two more overlays (spyglas and freezing).